File: app/schemas/equipment_profiles.py
# app/schemas/equipment_profiles.py

from typing import Optional

from pydantic import BaseModel, ConfigDict


class EquipmentProfileBase(BaseModel):
    manufacturer: str
    model: str
    category: str

    description: Optional[str] = None
    specifications: Optional[str] = None

    best_for: Optional[str] = None
    not_for: Optional[str] = None
    key_benefits: Optional[str] = None
    common_uses: Optional[str] = None
    faq: Optional[str] = None
    comparison_notes: Optional[str] = None

    manufacturer_url: Optional[str] = None
    source_notes: Optional[str] = None


class EquipmentProfileCreate(EquipmentProfileBase):
    pass


class EquipmentProfileUpdate(BaseModel):
    manufacturer: Optional[str] = None
    model: Optional[str] = None
    category: Optional[str] = None

    description: Optional[str] = None
    specifications: Optional[str] = None

    best_for: Optional[str] = None
    not_for: Optional[str] = None
    key_benefits: Optional[str] = None
    common_uses: Optional[str] = None
    faq: Optional[str] = None
    comparison_notes: Optional[str] = None

    manufacturer_url: Optional[str] = None
    source_notes: Optional[str] = None


class EquipmentProfileResponse(EquipmentProfileBase):
    model_config = ConfigDict(from_attributes=True)

    id: int

File: front3/src/types/equipmentProfiles.ts
// front3/src/types/equipmentProfiles.ts

export interface EquipmentProfile {
  id: number;
  manufacturer: string;
  model: string;
  category: string;
  description?: string | null;
  specifications?: string | null;
  best_for?: string | null;
  not_for?: string | null;
  key_benefits?: string | null;
  common_uses?: string | null;
  faq?: string | null;
  comparison_notes?: string | null;
  manufacturer_url?: string | null;
  source_notes?: string | null;
}

export interface EquipmentProfileCreate {
  manufacturer: string;
  model: string;
  category: string;
}

export interface EquipmentProfileUpdate {
  manufacturer?: string;
  model?: string;
  category?: string;
  description?: string | null;
  specifications?: string | null;
  best_for?: string | null;
  not_for?: string | null;
  key_benefits?: string | null;
  common_uses?: string | null;
  faq?: string | null;
  comparison_notes?: string | null;
  manufacturer_url?: string | null;
  source_notes?: string | null;
}